src/main/java/tasks/TimedTask.java

```diff
@@ -0,0 +1,91 @@
+package tasks;
+
+import exceptions.DukeDateTimeException;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Period;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+
+/**
+ * TimedTask is a abstract class that inherits from the base Task
+ * class, to add a new field of datetime into this child class which implements such functionalities
+ */
+abstract class TimedTask extends Task {
+    protected static final DateTimeFormatter FMAT;
+    protected static final LocalDate NOW = LocalDateTime.now().toLocalDate();
+    static {
+        DateTimeFormatterBuilder dateTimeFormatterBuilder = new DateTimeFormatterBuilder();
+        dateTimeFormatterBuilder.appendPattern("dd-MM-yyyy");
+        dateTimeFormatterBuilder.parseDefaulting(ChronoField.SECOND_OF_MINUTE, 0);
```

Comment:
The use of ChronoField is new to me, thanks for showing this alternative way to handle Date/Time :+1:

Reply:
Sorry the review was done a while ago and i din respond, but thanks for the review! Glad to know you learnt something!